Automattic/sensei
# Scope composer dependencies

Resolves #6574

## Proposed Changes

* Use `php-scoper` to scope the composer dependencies by moving them to a new namespace (`Sensei\ThirdParty`).
* The scoped dependencies live in a new folder (`third-party`).
* Use the standard composer autoloader to load the scoped dependencies. This approach avoids some of the autoloading issues described in https://github.com/humbug/php-scoper/issues/442.
* Update the CI workflows to use the scoped dependencies.

## Testing Instructions


I was not able to add tests for the composer `Actions` class. The passed Event class is only loaded by composer so I can't mock it. Let me know if you have suggestions on how to test it.

1. Delete your `vendor` folder.
2. Run `composer install`.
3. Check that the `third-party` folder is created and not empty.
4. Remove the dev dependencies by running `composer install --no-dev`
5. Check that your `vendor` folder has only the composer autoloader related files and no dev dependencies.
6. Enable the `email_customization` feature flag.
7. Go to Sensei LMS -> Settings -> Emails and click "Preview" on one of the emails. This makes sure the `emogrifier` dependency is loaded and working.
8. Download the CI generated build zip and make sure it works.
9. Make sure the composer dev dependencies are not included in the CI generated build zip.

## Pre-Merge Checklist
